lib/Cabal2Nix.hs

```diff
@@ -94,7 +95,6 @@ shakeBranch :: (Foldable t, Foldable f) => CondBranch v (t c) (f a) -> Maybe (Co
 shakeBranch (CondBranch c t f) = case (shakeTree t, f >>= shakeTree) of
   (Nothing, Nothing) -> Nothing
   (Nothing, Just f') -> shakeBranch (CondBranch (CNot c) f' Nothing)
-  (Just (CondNode d _c [(CondBranch c' t' f')]), Nothing) | null d -> Just $ CondBranch (CAnd c c') t' f'
```

Comment:
@angerman Oh, I removed this line because it was clearly causing things to be mistranslated. IIRC, there was some package that depended on `Win32` conditionally that this broke, causing it to *always* depend on it. Looking at this line, it seems wrong, but I admit I don't really fully understand this function.

Reply:
@ElvishJerricco this could absolutely be wrong. The idea is to simplify the tree and drop unnecessary branching. 

So assuming we are in a conditional branch, and clean up both `t`rue and `f`alse subtrees,
now we have only the `t`rue side, and nothing left for the `f`alse side. We should be able to ignore this, and move the conditional with an AND one node down.

Before shaking:
```
- c - <true sub tree: t>
  '-- <false sub tree: f>
```
after shaking:
```
- c - <some node with no interesting data, but a another branch>
  '-- Nothing
```
so this looks a bit like this:
```
- c - c' - <true subtree: t'>
  |   '--- <false subtree: f'>
  '------- Nothing
```

So we try to simplify this as
```
- (c && c') - <true subtree: t'>
      '------ <false subtree: f'>
```

And this is where the logic is wrong I guess. If `c` is `false`, we don't want `f'`, however the simplification would yield that instead of nothing.